examples/ticks_and_spines/date_precision_and_epochs.py

```diff
@@ -0,0 +1,152 @@
+"""
+=========================
+Date Precision and Epochs
+=========================
+
+Matplotlib can handle `.datetime` objects and `numpy.datetime64` objects using
+a unit converter that recognizes these dates and converts them to floating
+point numbers.
+
+Before Matplotlib 3.3, the default for this conversion returns a float that was
+days since "0000-01-01T00:00:00".  As of Matplotlib 3.3, the default is
+days from "2000-01-01T00:00:00".  This allows more resolution for modern
+dates.  "2000-01-01" with the old epoch converted to 730120, and a 64-bit
+floating point number has a resolution of 2^{-52}, or approximately
+14 microseconds, so microsecond precision was lost.  With the new default
+epoch "2020-01-01" is 7306.0, so the achievable resolution is 0.14
+microseconds.
+
+"""
+import datetime
+import dateutil
+import numpy as np
+
+import matplotlib
+import matplotlib.pyplot as plt
+import matplotlib.dates as mdates
+
+#############################################################################
+# Datetime
+# --------
+#
+# Python `.datetime` objects have microsecond reesolution, so with the
+# old default matplotlib dates could not round-trip full-resolution datetime
+# objects.
+
+mdates._reset_epoch()  # Don't do this.  Just being done for this tutorial.
+mdates.set_epoch('0001-01-01T00:00:00')  # old epoch (pre MPL 3.3)
```

Comment:
Let's get this right; see above for the actual epoch.

Reply:
Fixed...